pkg/util/clusterdata/worker_profiles_task.go

```diff
@@ -78,7 +86,10 @@ func (ef *workerProfilesEnricherTask) FetchData(callbacks chan<- func(), errs ch
 
 		machineProviderSpec, ok := o.(*azureproviderv1beta1.AzureMachineProviderSpec)
 		if !ok {
-			ef.log.Errorf("failed to read provider spec from the machine set %q: %T", machineset.Name, o)
+			// This should never happen: codecs uses scheme that has only one registered type
+			// and if something is wrong with the provider spec - decoding should fail
+			err := fmt.Errorf("failed to read provider spec from the machine set %q: %T", machineset.Name, o)
+			ef.log.Errorf(err.Error())
 			errs <- err
```

Comment:
This change is not related to the panic we are fixing in this PR. And it shold not be a problem as it is impossible (I think) to hit this code path (hence no tests for it).

However if we were to hit it, without the change we would send a nil error to the `errs` channel here.